crates/autopilot/src/run.rs

```diff
@@ -173,17 +173,22 @@ pub async fn run(args: Arguments, shutdown_controller: ShutdownController) {
         .await
         .unwrap();
 
-    let db_read = if let Some(db_read_url) = args.db_read_url
+    let (db_read, run_analyze) = if let Some(db_read_url) = args.db_read_url
         && args.db_write_url != db_read_url
     {
-        Postgres::new(db_read_url.as_str(), args.insert_batch_size)
-            .await
-            .expect("failed to create read replica database")
+        (
+            Postgres::new(db_read_url.as_str(), args.insert_batch_size)
+                .await
+                .expect("failed to create read replica database"),
+            // If the DB is in read-only mode, running ANALYZE is not possible and will trigger and
+            // error https://www.postgresql.org/docs/current/hot-standby.html
+            false,
+        )
     } else {
-        db_write.clone()
+        (db_write.clone(), true)
     };
```

Comment:
Since we now have the read replica configured everywhere this change will cause the metrics to not be updated anymore. As it is right now this seems like a footgun.
If it works with the write_db we should always pass that instead of silently "disabling" this functionality when the read_db is configured.